src/Roslyn/Core/ApiDesign/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -71,14 +71,23 @@ internal Impl(ApiData shippedData, ApiData unshippedData)
             {
                 _unshippedData = unshippedData;
 
+                // We ignore any duplicate entries here. These will be found and reported
+                // as diagnostics by DeclarePublicAPIAnalyzer.ValidateApiFiles.
```

Comment:
It seems analysis wouldn't even get here if duplicate symbols are found in api files.